flow/connectors/mysql/mysql.go

```diff
@@ -114,6 +128,119 @@ func (c *MySqlConnector) Execute(ctx context.Context, cmd string, args ...interf
 	return nil, connectionErr
 }
 
+func (c *MySqlConnector) ExecuteSelectStreaming(ctx context.Context, cmd string, result *mysql.Result,
+	rowCb client.SelectPerRowCallback,
+	resultCb client.SelectPerResultCallback,
+	args ...interface{},
+) error {
+	var connectionErr error
+	for conn, err := range c.withRetries(ctx) {
+		if err != nil {
+			return err
+		}
+
+		if len(args) == 0 {
+			if err := conn.ExecuteSelectStreaming(cmd, result, rowCb, resultCb); err != nil {
+				if mysql.ErrorEqual(err, mysql.ErrBadConn) {
+					connectionErr = err
+					continue
+				}
+				return err
+			}
+		} else {
+			stmt, err := conn.Prepare(cmd)
+			if err != nil {
+				if mysql.ErrorEqual(err, mysql.ErrBadConn) {
+					connectionErr = err
+					continue
+				}
+				return err
+			}
+			err = stmt.ExecuteSelectStreaming(result, rowCb, resultCb, args...)
+			_ = stmt.Close()
+			if err != nil {
+				if mysql.ErrorEqual(err, mysql.ErrBadConn) {
+					connectionErr = err
+					continue
+				}
+				return err
+			}
+		}
+		return nil
+	}
+	return connectionErr
+}
+
+func (c *MySqlConnector) GetGtidModeOn(ctx context.Context) (bool, error) {
+	if c.Flavor() == mysql.MySQLFlavor {
+		rr, err := c.Execute(ctx, "select @@gtid_mode")
+		if err != nil {
+			return false, err
+		}
+
+		gtid_mode, err := rr.GetString(0, 0)
+		if err != nil {
+			return false, err
+		}
+
+		return gtid_mode == "ON", nil
+	} else {
+		// mariadb always enabled: https://mariadb.com/kb/en/gtid/#using-global-transaction-ids
+		return true, nil
```

Comment:
nit: it looks like MariaDB GTID was added in `10.0.2` https://mariadb.com/kb/en/mariadb-1002-release-notes/
very old version, I'm fine with having validation reject this version and also older MySQL versions

Reply:
Decision blocked on proper support for GTID & MariaDB, don't want that to block this PR. Could have this GTID function always return false for now